### DISCUSSION TITLE
How do I generate the 'Design System' content?

### DISCUSSION THREAD

**Discussion by victorlmneves** [histoire-dev/histoire#125] [Q&A]
Is the Design System menu available on the demo set up on the config?
![image](https://user-images.githubusercontent.com/17341789/172237275-f46f6039-fdf4-42ab-b7ed-6e5ed0cc49e6.png)

**Comment by hugoattal** [comment]
All the examples are available here: https://github.com/histoire-dev/histoire/tree/main/examples

The design system section is generated by a built-in tailwind plugin that checks for a tailwind configuration file like this:
```
'tailwind.config.js',
'tailwind.config.cjs',
'tailwind.config.mjs',
'tailwind.config.ts',
'tailwind-config.js',
'tailwind-config.cjs',
'tailwind-config.mjs',
'tailwind-config.ts',
```

This example uses this file: https://github.com/histoire-dev/histoire/blob/main/tailwind.config.cjs

(We should document this, I don't think it's in the doc yet 🤔...)

**Reply by victorlmneves** [reply]
@hugoattal If it is I didn't see or didn't search well :D
I was asking because I'm not using tailwind or any other CSS framework, but we use StyleDictionary where we get all the tokens and, my idea is to generate something like that, but based on StyleDictionary tokens.
Thanks

**Reply by hugoattal** [reply]
If you want to build a plugin like this for StyleDictionary, you can take a look at the tailwind plugin source here 🙂
https://github.com/histoire-dev/histoire/blob/main/packages/histoire/src/node/builtin-plugins/tailwind-tokens.ts

Also, here is the plugin creation documentation here: https://histoire.dev/guide/plugins.html#plugin-development
